### PR TITLE
setup cob4-4

### PR DESCRIPTION
**Pull request by NadiaHG**

### CODE REVIEW

File: cob_bringup/controllers/base_controller_plugin.launch [hunk 1]
```diff
@@ -14,7 +14,7 @@
 		<node pkg="controller_manager" type="spawner" name="joint_state_controller_spawner" args="--timeout 60 --shutdown-timeout 1 joint_state_controller" respawn="true" />
 		<node pkg="controller_manager" type="spawner" name="twist_controller_spawner" args="--timeout 60 --shutdown-timeout 1 twist_controller" respawn="true" output="screen"/>
 		<node pkg="controller_manager" type="spawner" name="odom_controller_spawner" args="--timeout 60 --shutdown-timeout 1 odometry_controller" respawn="true" output="screen"/>
-
+		<!--node pkg="controller_manager" type="spawner" name="velocity_controller_spawner" args="-/-timeout 60 -/-shutdown-timeout 1 joint_group_velocity_controller" respawn="true" output="screen"/-->
```
Comment (fmessmer) [inline cob_bringup/controllers/base_controller_plugin.launch:17]: why is this comment added?

File: cob_bringup/tools/teleop.launch [hunk 2]
```diff
@@ -4,10 +4,14 @@
 	<arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
 	<arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
 	<arg name="version" default="v2"/>
```
Comment (fmessmer) [inline cob_bringup/tools/teleop.launch:6]: why do we set default to `v2` if every robot uses `v1`?

File: cob_bringup/tools/teleop_v1.xml [hunk 3]
```diff
@@ -10,8 +12,10 @@
 	</include>
 
 	<node pkg="cob_teleop" type="cob_teleop_v1" name="$(anon teleop_node)" ns="teleop" output="screen">
```
Comment (fmessmer) [inline cob_bringup/tools/teleop_v1.xml:14]: order: ns, pkg, type, name...

File: cob_bringup/tools/teleop_v1.xml [hunk 4]
```diff
@@ -10,8 +12,10 @@
...
-		<remap from="/cmd_vel" to="/base_controller/command" />
+		<remap from="/cmd_vel" to="$(arg cmd_vel)" />
 		<remap from="/cmd_vel_safety" to="/base_controller/command_safe" />
+		<remap from="/base_controller/init" to="$(arg service_ns)/init" />
+		<remap from="/base_controller/recover" to="$(arg service_ns)/recover" />
 		<rosparam command="load" ns="/robot_config" file="$(arg pkg_hardware_config)/$(arg robot)/config/robot_modules.yaml"/>
 		<rosparam command="load" ns="/teleop" file="$(arg pkg_hardware_config)/$(arg robot)/config/teleop.yaml"/>
```
Comment (fmessmer) [inline cob_bringup/tools/teleop_v1.xml:20]: order: ns, command file

why is there always a `/` at the beginning...?

btw. should we put the whole thing into a <group ns="teleop"></group>?

File: cob_bringup/tools/teleop_v2.xml [hunk 5]
```diff
@@ -10,7 +12,9 @@
 	</include>
 
 	<node pkg="cob_teleop" type="cob_teleop_v2" name="$(anon teleop_node)" output="screen" >
-		<remap from="base_controller_command" to="base/twist_controller/command"/>
+		<remap from="/base/driver/init" to="$(arg service_ns)/init"/>
+		<remap from="/base/driver/recover" to="$(arg service_ns)/recover"/>
```
Comment (fmessmer) [inline cob_bringup/tools/teleop_v2.xml:16]: should we not simply 
`<remap from="init" to="$(arg service_ns)/init"/>`
maybe we need to change the implementation of the teleop nodes...

I guess all topics used in teleop nodes should be relative..i.e. only `init` or `recover``or``cmd_vel``

File: cob_hardware_config/cob4-4/config/teleop.yaml [hunk 6]
```diff
@@ -3,13 +3,15 @@ run_factor: 2.0
 
 # buttons
 deadman_button: 5
-run_button: 7
+#run_button: 7
 base_safety_button: 4
+recover_base_button: 7
```
Comment (fmessmer) [inline cob_hardware_config/cob4-4/config/teleop.yaml:8]: this changes the meaning of the buttons, right?
is this only special for cob4-4 or is this something to be used on the other robots, too?

**Comment by fmessmer** [comment]
I think we should have a closer look at the implementation of the teleop nodes...in order to check for possible simplifications...e.g. namespaces and remaps...

also, now that we have a cob3-2 functional in indigo...should we update the launch and yaml files for cob3-6 and cob3-9 according to our new insights...to make it consistent with cob3-2 again?

**Comment by floweisshardt** [comment]
cob_teleop issues will be resolved in https://github.com/ipa320/cob_command_tools/issues/92

**Comment by floweisshardt** [comment]
wait for https://github.com/ipa320/cob_robots/issues/311 to be resolved

**Comment by fmessmer** [comment]
What's the status here?

**Comment by floweisshardt** [comment]
@ipa-nhg please resolve merge conflicts and update this PR (please do not open a new one)

**Comment by floweisshardt** [comment]
superceeded by #332, closing

**Pull request closed without merging**